### ISSUE TITLE
Advocacy document motivating automatic formatting

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/fmt-rfcs#54]
We should have a document, probably here and in the rustfmt repo, and hopefully blessed by the wider Rust community (doc team? core team? wider support somehow?) in some way, that advocates for the principle of automatic formatting and for doing so using the defaults rather than customising. Some maintainers may need persuading that using Rustfmt is beneficial and we should lay out the benefits in detail.

**Comment by joshtriplett** [comment]
@nrc Something laying out the arguments for uniformity, ease of contribution across many projects, new-developer friendliness, etc?

**Comment by nrc** [comment]
yes

**Comment by johnthagen** [comment]
This issue made me think of a nice [Google presentation](https://www.youtube.com/watch?v=uvddFPavYZQ#t=19m24s) about the importance they placed in ClangFormat.

TLDR:

* Humans understand through patterns, so consistent formatting helps humans understand the code more quickly.
* Debating formatting can consume a lot of time. Ideally community consensus and tools can cut down on the time each individual project/team has to repeat spending this time.
* Refactoring code and having to subsequently fix whitespace costs non-trivial amounts of time for the developer.

**Comment by phaylon** [comment]
I would like to propose also adding a small note to respect if a project made a choice deviating from the defaults.
